src/styles/index.scss

```diff
@@ -0,0 +1,101 @@
+/**
+ * Naming scheme for variables is `--component-name-size-property-modifier`.
+ */
+
+@import './variables.scss';
+
+:root {
+  /* Colors */
```

Comment:
So I guess the idea here is that

- Component-specific colors, like `--button-background-color-hover` default to GDS colors but may be overwritten by the site using GDS.
- GDS-specific colors like `ui-01` are just used as the default value for the component-specific colors but cannot be overwritten by the site.

?

Reply:
> GDS-specific colors like ui-01 are just used as the default value for the component-specific colors but cannot be overwritten by the site.

Why shouldnt they be overwritten by the site? I think it makes sense to change them "globally".

Reply:
Hmm, I think that would work as well. My thinking was that these (ui-01 etc.) refer to Genero's brand colors and therefore any site besides Genero's internal sites shouldn't really care about them. Like if instead of "ui-01" the variable would be called "genero-brand-bright-orange" it wouldn't make sense to modify that to anything besides the original color. Instead the site would care about defining a primary and secondary color etc. directly.

But you're probably right, I guess they might have these generic names specifically so that they may be modified to anything. I do still think it's a bit cumbersome that a site must have seven different UI colors that are arbitrarily mapped to various UI components, but probably it will work OK in practice.

Reply:
> I do still think it's a bit cumbersome that a site must have seven different UI colors that are arbitrarily mapped to various UI components, but probably it will work OK in practice.

Yeah, Carbon Design System did look pretty confusing with all those generic names not saying anything about the colors. But they did have it set up pretty much like this too.

Reply:
We can probably iterate on it later too. I especially think that the gray colors could have their own naming to distinguish them from "accent" colors that are used mostly for buttons/backgrounds. Not sure what happens if it's ever common with shades of accent colors.